core/src/main/java/org/apache/gravitino/EntityStore.java

```diff
@@ -55,15 +55,19 @@ public interface EntityStore extends Closeable {
    * <p>Note. Depends on the isolation levels provided by the underlying storage, the returned list
    * may not be consistent.
    *
-   * @param namespace the namespace of the entities
    * @param <E> class of the entity
+   * @param namespace the namespace of the entities
    * @param type the detailed type of the entity
    * @param entityType the general type of the entity
-   * @throws IOException if the list operation fails
+   * @param allowMissingFields Some fields may have a relatively high acquisition cost, EntityStore
+   *     provide an optional setting to avoid fetching these high-cost fields to improve the
+   *     performance.
    * @return the list of entities
+   * @throws IOException if the list operation fails
    */
   <E extends Entity & HasIdentifier> List<E> list(
-      Namespace namespace, Class<E> type, EntityType entityType) throws IOException;
+      Namespace namespace, Class<E> type, EntityType entityType, List<Field> allowMissingFields)
+      throws IOException;
```

Comment:
Maybe we can add a another inteface with default to fetch all the fields, then you don't have to change the code above for different catalogs.

Reply:
OK, I preferring adding another interface. cc @yuqi1129 @xunliu

Reply:
Also, I would suggest to rename the parameter to like: "List<Field> skippingFields"

Reply:
ok for me.